Missing hooks + Hook aliases as title (better search results in Algolia) (#1733)
PrestaShop/docs: README.md

## License

Content from this documentation is licensed under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).

## Maintaining Hook list referential

A complete referential of hook and informations on each hook is located in `modules/concepts/hooks/list-of-hooks/`. 

> This referential can be generated with a script, crawling all: 

Reviewer: A script that we can put somewhere to the repo?

Reviewer: as we spoke, it could be added as a Gist

Author: https://gist.github.com/thomasnares/b0605fdd9f5e77fa16bf5a42408feba3

I need to do some rewrite to fit the new format

Reviewer: let's add a task to link `with a script` to this file once it's rewritten to the new format, ok?

Author: yes 👍

> - Prestashop (core) codebase

Reviewer: ```suggestion
- PrestaShop (core) codebase
```

- all native modules
- all native themes

> In the documentation, a hook is described with in yaml format, with this format: 

Reviewer: ```suggestion
In the documentation, a hook is described in yaml format: 
```

```yaml
---
Title: string #name of the hook
hidden: boolean[true] #must be true
hookTitle: string #title for the description
description: string #description sentence
origin: string #core, module or theme
files:
    -
        theme: string #theme name only if origin=theme
        module: string #module name only if origin=module
        url: string #url of the file on github
        file: string #path of the file
locations:
    - string #back office, front office or both
type: string #action or display
hookAliases:
    - string #old hook name or alias
array_return: boolean #true or false if the hook has an `$array_return` parameter set to `true`
check_exceptions: boolean #true or false if the hook has a `$check_exceptions` parameter set to `false`
chain: boolean #true or false if the hook has a `$chain` parameter set to `true` 
---
```

> This `yaml` structure allows the shortcode `hookDescriptor` to render a digest html page for the documentation, and allows the search script to search in this metadata. 

Reviewer: ```suggestion
This YAML structure enables the shortcode `hookDescriptor` to generate an HTML page with condensed information for documentation, and allows the search script to search within this metadata.
```

Next, add the shortcode in the page's content: 

```
{{% hookDescriptor %}}
```

> Finally, if available, add a section with a code example, and parameters details: 

Reviewer: ```suggestion
Finally, if available, add a section with a code example and parameters details: 
```
